### PR TITLE
Add check for device timestamp jump to node

### LINKED ISSUE #44: Detect time jump and reset

### PR DESCRIPTION
**Pull request by ykoga-kyutech**
close #44

### CODE REVIEW

**Comment by ykoga-kyutech** [comment]
@at-wat PTAL

**Comment by ykoga-kyutech** [comment]
Sure :+1: I will add soon.

File: include/device_time_origin.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  {
+  }
+  DriftedTime(const ros::Time origin, const float gain)
+    : origin_(origin)
+    , gain_(gain)
+  {
+  }
+};
+
+class DeviceTimeOriginEstimator
```
[the remark this answers is not in the document]
Comment (ykoga-kyutech) [reply include/device_time_origin.h]: changed to namespace `device_time_origin::estimator`

File: include/device_time_origin.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      const ros::Duration& communication_delay,
+      ros::Time& time_at_device_timestamp)
+  {
+    time_at_device_timestamp = ros::Time::fromBoost(time_response) - communication_delay * 0.5;
+
+    return time_at_device_timestamp - ros::Duration().fromNSec(device_timestamp * 1e6);
+  }
+};
+
+class DeviceTimeOriginJumpDetector
```
[the remark this answers is not in the document]
Comment (ykoga-kyutech) [reply include/device_time_origin.h]: changed to namespace `device_time_origin::jump_detector`

**Comment by ykoga-kyutech** [comment]
@at-wat PTAL again. 
I added namespace `device_time_origin` to entire device_time_origin.h

File: include/device_time_origin.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  inline DriftedTime(const ros::Time origin, const float gain)
+    : origin_(origin)
+    , gain_(gain)
+  {
+  }
+};
+
+namespace estimator
+{
+inline static ros::Time estimateOriginByAverage(
```
Comment (at-wat) [inline include/device_time_origin.h:45]: static (internal linkage) and inline are redundant.
```suggestion
inline ros::Time estimateOriginByAverage(
```

Comment (ykoga-kyutech) [reply include/device_time_origin.h]: done https://github.com/seqsense/urg_stamped/pull/66/commits/7f9f216d067c984fa1a2849f565a6e7d9d90d106

**Comment by ykoga-kyutech** [comment]
@at-wat PTAL again

**Pull request merged**